server/protocols/jmap/src/main/java/org/apache/james/jmap/model/message/MimePart.java

```diff
@@ -260,4 +261,23 @@ public HeaderCollection getHeaderCollection() {
                 .flatMap((mimePart) -> Stream.concat(Stream.of(mimePart), mimePart.getAttachmentsStream()));
     }
 
+    private static class MimePartComparator implements Comparator<MimePart> {
```

Comment:
Come on, I have to read the whole class to understand what it is supposed to do ...
You don't even have full unit test coverage, it's not ok